ReactNativeClient/lib/models/Setting.js

```diff
@@ -652,17 +656,32 @@ class Setting extends BaseModel {
 		return this.modelSelectAll('SELECT * FROM settings').then(rows => {
 			this.cache_ = [];
 
-			for (let i = 0; i < rows.length; i++) {
-				const c = rows[i];
+			const pushItemsToCache = (items) => {
+				for (let i = 0; i < items.length; i++) {
+					const c = items[i];
 
-				if (!this.keyExists(c.key)) continue;
-				c.value = this.formatValue(c.key, c.value);
-				c.value = this.filterValue(c.key, c.value);
+					if (!this.keyExists(c.key)) continue;
 
-				this.cache_.push(c);
-			}
+					c.value = this.formatValue(c.key, c.value);
+					c.value = this.filterValue(c.key, c.value);
+
+					this.cache_.push(c);
+				}
+			};
 
-			this.dispatchUpdateAll();
+			pushItemsToCache(rows);
+
+			shim.loadSecureItems(this.constants_.appId).then(secureItems => {
+				// secureItems is an array of { account: 'foo', password: 'bar' }
+				// but each item should be in the form of { key: 'foo', value: 'bar' }
+				// https://stackoverflow.com/a/50101979
```

Comment:
Could you abstract this away and make `shim.loadSecureItems` return an array of `{ key: 'foo', value: 'bar' }`?

Reply:
Sure, will do it.